x-pack/plugins/security_solution/common/endpoint/data_generators/endpoint_action_generator.ts

```diff
@@ -239,6 +241,24 @@ export class EndpointActionGenerator extends BaseDataGenerator {
       }
     }
 
+    if (command === 'upload') {
+      if (!details.parameters) {
+        (
+          details as ActionDetails<
```

Comment:
Do we need the cast?

Reply:
Yes because in order to set the parameters below, the `ActionDetails` must be an `upload` action details, so the cast here using the output content and parameters type does that.